packages/typespec-python/package.json

```diff
@@ -73,6 +73,13 @@
         "eslint": "^8.44.0",
         "mocha": "~10.2.0",
         "rimraf": "~5.0.0",
-        "typescript": "~5.1.3"
+        "typescript": "~5.1.3",
+        
+        "@azure-tools/typespec-azure-core": "~0.35.1",
```

Comment:
why do we need to add these dependencies outside of peer dependencies? And shouldn't we keep them floating as well

Reply:
We want to uptake a pattern of having the emitter float its peer dependencies, but pinning the version of the peers as dev dependencies in emitter-package.json.

```json
{
  "main": "dist/src/index.js",
  "dependencies": {
    "@azure-tools/typespec-python": "0.15.10"
  },
  "devDependencies": {
    "@azure-tools/typespec-client-generator-core": "0.36.0-dev.8",
    "@typespec/compiler": "~0.49.0",
    "@typespec/http": "~0.49.0",
    "@typespec/rest": "~0.49.0",
    "@typespec/versioning": "~0.49.0",
    "@azure-tools/typespec-azure-core": "~0.35.1"
  }
}
```

This will allow us to commit an emitter-package-lock.json along with emitter-package.json to improve ci build security.   We generate emitter-package.json from the emitter's package.json file using this script in azure-sdk-tools: [New-EmitterPackageJson.ps1](https://github.com/Azure/azure-sdk-tools/blob/main/eng/scripts/autorest/New-EmitterPackageJson.ps1)